omegaconf/omegaconf.py

```diff
@@ -906,6 +906,9 @@ def _node_wrap(
     )
     if is_dict:
         key_type, element_type = get_dict_key_value_types(type_)
+        flags = {}
+        if is_dict_annotation(type_):
+            flags["struct"] = False
```

Comment:
Not directly related to this change, but I was trying to understand how things are working, and I'm confused by this:

```python
cfg = OmegaConf.create({"a": {"b": 0}})
OmegaConf.set_struct(cfg, True)
cfg.a = {"b": 1}
```
which raises this exception:
```
ConfigKeyError: Key 'b' is not in struct
    full_key: a.b
    object_type=dict
```
What's going on there? (it's the same on master btw)

Reply:
I would call it a bug.
file an issue. I am expecting this assignment to succeed.